src/ILCompiler.Compiler/src/Compiler/DependencyAnalysis/CodeBasedDependencyAlgorithm.cs

```diff
@@ -67,6 +67,15 @@ public static void AddDependenciesDueToMethodCodePresence(ref DependencyList dep
                         "UniversalCanon signature of method"));
                 }
 
+                if (method.IsVirtual && !method.HasInstantiation)
```

Comment:
Oh, right. I think I remember why. It's a workaround for not having a `ReflectableMethodNode`. Since we don't have a pressing need for a workaround anymore and I'm actually working on a `ReflectableMethodNode`, let's not check this part (`CodeBasedDependencyAlgorithm.cs`) in.